src/meshwifi/meshwifi.cpp

```diff
@@ -0,0 +1,212 @@
+            break;
+        case SYSTEM_EVENT_STA_START:
+            DEBUG_MSG("WiFi client started");
+            break;
+        case SYSTEM_EVENT_STA_STOP:
+            DEBUG_MSG("WiFi clients stopped");
+            break;
+        case SYSTEM_EVENT_STA_CONNECTED:
+            DEBUG_MSG("Connected to access point");
+            break;
+        case SYSTEM_EVENT_STA_DISCONNECTED:
+            DEBUG_MSG("Disconnected from WiFi access point");
+
+            // Reconnect WiFi
+            reconnectWiFi();
+            break;
+        case SYSTEM_EVENT_STA_AUTHMODE_CHANGE:
+            DEBUG_MSG("Authentication mode of access point has changed");
+            break;
+        case SYSTEM_EVENT_STA_GOT_IP:
+            DEBUG_MSG("Obtained IP address: ");
+            Serial.println(WiFi.localIP());
+
+            // Start web server
+            initWebServer();
+            break;
+        case SYSTEM_EVENT_STA_LOST_IP:
+            DEBUG_MSG("Lost IP address and IP address is reset to 0");
+            break;
+        case SYSTEM_EVENT_STA_WPS_ER_SUCCESS:
+            DEBUG_MSG("WiFi Protected Setup (WPS): succeeded in enrollee mode");
+            break;
+        case SYSTEM_EVENT_STA_WPS_ER_FAILED:
+            DEBUG_MSG("WiFi Protected Setup (WPS): failed in enrollee mode");
+            break;
+        case SYSTEM_EVENT_STA_WPS_ER_TIMEOUT:
+            DEBUG_MSG("WiFi Protected Setup (WPS): timeout in enrollee mode");
+            break;
+        case SYSTEM_EVENT_STA_WPS_ER_PIN:
+            DEBUG_MSG("WiFi Protected Setup (WPS): pin code in enrollee mode");
+            break;
+        case SYSTEM_EVENT_AP_START:
+            DEBUG_MSG("WiFi access point started");
+            break;
+        case SYSTEM_EVENT_AP_STOP:
+            DEBUG_MSG("WiFi access point stopped");
+            break;
+        case SYSTEM_EVENT_AP_STACONNECTED:
+            DEBUG_MSG("Client connected");
+            break;
+        case SYSTEM_EVENT_AP_STADISCONNECTED:
+            DEBUG_MSG("Client disconnected");
+            break;
+        case SYSTEM_EVENT_AP_STAIPASSIGNED:
+            DEBUG_MSG("Assigned IP address to client");
+            break;
+        case SYSTEM_EVENT_AP_PROBEREQRECVED:
+            DEBUG_MSG("Received probe request");
+            break;
+        case SYSTEM_EVENT_GOT_IP6:
+            DEBUG_MSG("IPv6 is preferred");
+            break;
+        case SYSTEM_EVENT_ETH_START:
+            DEBUG_MSG("Ethernet started");
+            break;
+        case SYSTEM_EVENT_ETH_STOP:
+            DEBUG_MSG("Ethernet stopped");
+            break;
+        case SYSTEM_EVENT_ETH_CONNECTED:
+            DEBUG_MSG("Ethernet connected");
+            break;
+        case SYSTEM_EVENT_ETH_DISCONNECTED:
+            DEBUG_MSG("Ethernet disconnected");
+            break;
+        case SYSTEM_EVENT_ETH_GOT_IP:
+            DEBUG_MSG("Obtained IP address");
+            break;
+        default: break;
+    }
+}
+
+void handleJSONChatHistory() {
+
```

Comment:
hmm - perhaps we should leave this in a feature branch until this example code is either disabled or more real?

Reply:
i.e. we could leave this in a "wifidev" branch until the API is exposed on a port.

Reply:
Agreed. I've never made a feature branch in GitHub. I'll look into that.

Reply:
I'll do some further refactoring. Will split the WiFi radio handling from serving HTTP pages, then what can go into main would be a simple "Hello World" while the rest of the work goes into the wifidev feature branch.